Review followup: PR #21432 — t3001: gh-status-helper for incident detection + recovery runbook

## Unaddressed review bot suggestions

PR #21432 was merged with unaddressed review bot feedback. Each comment
below includes its file path, line number, a direct link to the inline
review comment, and a `diff` fence with the code context the bot was
flagging. Resolved and outdated threads are filtered out via GitHub's
GraphQL review-thread state. Read the relevant lines, decide whether
the suggestion is correct, and either apply the fix or close this issue
with a wontfix rationale.

**Source PR:** https://github.com/marcusquinn/aidevops/pull/21432

---

### You are the triager (worker-is-triager rule)

This issue is **auto-created from review bot output** and dispatched
directly to you. Review bots can be wrong: hallucinated line refs, false
premises about codebase structure, template-driven sweeps without
measurements (see GH#17832-17835 for prior art and `prompts/build.txt`
section 6a). **Do not assume the bot is correct.** Verify before acting.

You must end in exactly one of three outcomes — no fourth "hand it back
to the human" path exists. Humans approve decisions; they do not re-do
analysis.

#### Outcome A — Premise falsified → close the issue

1. Read the cited `file:line` (listed under *Files to modify* below).
2. If the bot's claim is factually wrong (file doesn't exist at that
   line, function doesn't behave as described, "auto-generated" section
   isn't actually auto-generated, etc.), **close the issue** with a
   comment in this shape:

   > **Premise falsified.** \<what the bot claimed\>. \<what the code
   > actually shows, with a `file:line` citation or one-line quote\>.
   > Not acting.

   No PR. No further dispatch. The closing comment trains the next
   session reading this thread and the noise filter.

#### Outcome B — Premise correct + fix is obvious → implement and PR

1. Verify the bot's premise as above.
2. Read the Worker Guidance section below, open a worktree, implement.
3. Open a PR with `Resolves #<this-issue-number>` in the body
   (use THIS issue's number, not the source PR's) so merge auto-closes it.
4. Follow the normal Lifecycle Gate (brief, tests, review-bot-gate,
   merge, postflight).

#### Outcome C — Premise correct but approach is a genuine judgment call

Only use this path if you reach it after Outcomes A and B don't apply:
the bot's finding is real, but the fix requires a decision that is
architectural, policy, breaking-change, or otherwise genuinely outside
what you can resolve autonomously. In that case, post a **decision
comment** with exactly these fields:

- **Premise check:** one line, confirming the finding is real.
- **Analysis:** 2-4 bullets on the trade-offs.
- **Recommended path:** the option you would take if the decision were
  yours, with rationale.
- **Specific question:** the single decision the human needs to make
  (yes/no or pick-one, not open-ended).

Then apply `needs-maintainer-review` and stop. The human wakes up to a
ready-to-approve recommendation, not a blank task.

> **Ambiguity about scope or style is not Outcome C.** Per
> `prompts/build.txt` "Reasoning responsibility", the model does the
> thinking and delivers a recommendation. Only escalate what is genuinely
> a maintainer-only decision.

### Worker Guidance

**Files to modify:**

- `.agents/scripts/gh-status-helper.sh:198`
- `.agents/scripts/gh-status-helper.sh:236`

**Implementation steps (Outcome B path):**

1. Read the `diff` block under each inline comment below — it shows the
   exact code the bot was flagging. Open the file only if you need
   surrounding context beyond what the diff tail shows.
2. Read the bot's full comment below the diff — it contains the rationale
   and any suggested change.
3. Verify the premise before implementing (see Outcome A). If the premise
   is wrong, switch to Outcome A instead of burning iterations trying to
   satisfy a wrong suggestion.
4. If multiple comments target the same file, group your edits into one
   logical commit.
5. Run `shellcheck` / `markdownlint-cli2` / project tests as appropriate.

**Verification:**

- Open the new PR with `Resolves #<this-issue>` so this followup is auto-closed on merge.
- If the bot's suggestion was incorrect, close this issue with a Outcome A comment — do not open a no-op PR.

### Inline comments

#### gemini-code-assist on `.agents/scripts/gh-status-helper.sh:198`

[View inline comment](https://github.com/marcusquinn/aidevops/pull/21432#discussion_r3150675732)

```diff
+cmd_incidents() {
+	if ! _ensure_cache "$INCIDENTS_URL" "$CACHE_INCIDENTS"; then
+		if [[ "$ARG_JSON" -eq 1 ]]; then
+			printf '{"incidents":[],"reason":"network_failure"}\n'
+		else
+			printf '%s — could not reach Statuspage API\n' "$INDICATOR_UNKNOWN" >&2
+		fi
+		return 3
+	fi
+
+	if [[ "$ARG_JSON" -eq 1 ]]; then
+		jq '{incidents: [.incidents[] | {name, impact, started_at, latest_update: (.incident_updates[0].body // ""), components: [.components[].name]}]}' "$CACHE_INCIDENTS"
```

> ![medium](https://www.gstatic.com/codereviewagent/medium-priority.svg)
> 
> Using `.incidents[]` will cause `jq` to error if the `.incidents` key is missing or null in the JSON response. Using the optional operator `.incidents[]?` is safer and ensures the function returns a valid structure even on empty results, preventing downstream pipe failures.
> 
> ```suggestion
> 		jq '{incidents: [.incidents[]? | {name, impact, started_at, latest_update: (.incident_updates[0].body // ""), components: [.components[].name]}]}' "$CACHE_INCIDENTS"
> ```
> 
> <details>
> <summary>References</summary>
> 
> 1. When a shell function is expected to return JSON, it should output a valid empty JSON array ([]) or object ({}) on no results to ensure consumers of the output do not break.
> </details>

#### gemini-code-assist on `.agents/scripts/gh-status-helper.sh:236`

[View inline comment](https://github.com/marcusquinn/aidevops/pull/21432#discussion_r3150675745)

```diff
+	timestamp=$(date -u +"%Y-%m-%dT%H:%M:%SZ")
+
+	printf '<!-- aidevops:gh-status-correlation -->\n'
+	# shellcheck disable=SC2016 # backticks are markdown, not command substitution
+	printf '**GitHub platform state at %s:** `%s` — %s\n\n' "$timestamp" "$indicator" "$description"
+
+	if [[ -f "$CACHE_INCIDENTS" ]]; then
+		local count
+		count=$(jq '.incidents | length' "$CACHE_INCIDENTS")
+		if [[ "$count" -gt 0 ]]; then
+			printf 'Active incidents:\n\n'
+			jq -r '.incidents[] | "- **[\(.impact)]** \(.name) — components: \([.components[].name] | join(", ")) (started \(.started_at))"' "$CACHE_INCIDENTS"
```

> ![medium](https://www.gstatic.com/codereviewagent/medium-priority.svg)
> 
> Similar to the incidents subcommand, using `.incidents[]?` here prevents potential `jq` errors if the Statuspage API returns a response where the incidents key is null or missing, ensuring the status display remains robust.
> 
> ```suggestion
> 			jq -r '.incidents[]? | "- **[\(.impact)]** \(.name) — components: \([.components[].name] | join(", ")) (started \(.started_at))"' "$CACHE_INCIDENTS"
> ```
> 
> <details>
> <summary>References</summary>
> 
> 1. When a shell function is expected to return JSON, it should output a valid empty JSON array ([]) or object ({}) on no results to ensure consumers of the output do not break.
> </details>

### PR review summaries

#### gemini-code-assist[bot] review summary

[View review](https://github.com/marcusquinn/aidevops/pull/21432#pullrequestreview-4184727227)

> ## Code Review
> 
> This pull request introduces an incident recovery runbook and a corresponding helper script, `gh-status-helper.sh`, designed to distinguish between GitHub platform incidents and local framework bugs. The script queries the GitHub Statuspage API, provides status classifications, and generates correlation comments for issues. A comprehensive test suite is also included. Feedback focuses on improving the robustness of `jq` processing within the helper script by using the optional operator to safely handle missing or null keys in API responses.
<!-- aidevops:sig -->
---
[aidevops.sh](https://aidevops.sh) v3.13.5 automated scan.

--- a/.agents/scripts/gh-status-helper.sh
+++ b/.agents/scripts/gh-status-helper.sh
@@ -195,7 +195,7 @@ cmd_incidents() {
 	fi
 
 	if [[ "$ARG_JSON" -eq 1 ]]; then
-		jq '{incidents: [.incidents[] | {name, impact, started_at, latest_update: (.incident_updates[0].body // ""), components: [.components[].name]}]}' "$CACHE_INCIDENTS"
+		jq '{incidents: [.incidents[]? | {name, impact, started_at, latest_update: (.incident_updates[0].body // ""), components: [.components[].name]}]}' "$CACHE_INCIDENTS"
 	else
 		# Human-readable listing. Empty list = healthy.
 		local count
@@ -233,7 +233,7 @@ cmd_correlate() {
 		count=$(jq '.incidents | length' "$CACHE_INCIDENTS")
 		if [[ "$count" -gt 0 ]]; then
 			printf 'Active incidents:\n\n'
-			jq -r '.incidents[] | "- **[\(.impact)]** \(.name) — components: \([.components[].name] | join(", ")) (started \(.started_at))"' "$CACHE_INCIDENTS"
+			jq -r '.incidents[]? | "- **[\(.impact)]** \(.name) — components: \([.components[].name] | join(", ")) (started \(.started_at))"' "$CACHE_INCIDENTS"
 			printf '\n'
 		fi
 	fi

